javalanglib/src/main/scala/java/lang/Integer.scala

```diff
@@ -139,6 +164,36 @@ object Integer {
   def toHexString(i: scala.Int): String = toStringBase(i, 16)
   def toOctalString(i: scala.Int): String = toStringBase(i, 8)
 
+  // Ported from https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/java/lang/IntegralToString.java
+  def toString(i: Int, radix: Int): String = {
+    if (radix == 10 || radix < java.lang.Character.MIN_RADIX || radix > java.lang.Character.MAX_RADIX) {
+      Integer.toString(i)
+    }
+    else {
+      var _i = i
+      var negative = false
+      if (_i < 0) {
+        negative = true
+      } else {
+        _i = -_i
+      }
+      val bufLen: Int = if (radix < 8) 33 else 12
+      val buf = Array.ofDim[Char](bufLen)
+      var cursor: Int = bufLen
+      do {
+        val q: Int = _i / radix
+        cursor -= 1
+        buf(cursor) = DIGITS(radix * q - _i)
+        _i = q
+      } while (_i != 0)
+      if (negative) {
+        cursor -= 1
+        buf(cursor) = '-'
+      }
+      String.valueOf(buf, cursor, bufLen - cursor)
+    }
+  }
+
```

Comment:
Have you considered using JavaScript's [`toString`][1]? As far as I can see, it does exactly what we want. To enable it, import `js.JSNumberOps.enableJSNumberOps`.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString